core/extension/gdnative_interface.cpp

```diff
@@ -678,6 +681,59 @@ static const char32_t *gdnative_string_operator_index_const(const GDNativeString
 	return &self->ptr()[p_index];
 }
 
+static void gdnative_string_operator_plus_eq_string(GDNativeStringPtr p_self, const GDNativeStringPtr p_b) {
+	String *self = (String *)p_self;
+	const String *b = (const String *)p_b;
+	*self += *b;
+}
+
+static void gdnative_string_operator_plus_eq_char(GDNativeStringPtr p_self, char32_t p_b) {
```

Comment:
Could we add the encoding in the name of the function ?

I guess for `char32_t` it is expected to be utf32, but for the cstr version it's easy to end up with a platform-dependent encoding if one doesn't think about it.

On top of that we have latin1/utf8/utf16/utf32/wide versions of other functions working with C string, so do you think we should do the same here ?